### PR TITLE
Changed install script to make it compatible for sh on linux

### PR DESCRIPTION
**Pull request by amp-powell**

### CODE REVIEW

**Comment by JohnSchmeichel** [comment]
Works with default options, but if I set AZURE_ARTIFACTS_CREDENTIAL_PROVIDER_VERSION then I get errors:

```sh
export AZURE_ARTIFACTS_CREDENTIAL_PROVIDER_VERSION=v1.1.0-alpha
sh -c "$(curl -fsSL https://raw.githubusercontent.com/microsoft/artifacts-credprovider/ba93b5218249ec3792470f9b2f48c91bd930a016/helpers/installcredprovider.sh)"
```

still gives me an error `sh: 19: [: v1.1.0-alpha: unexpected operator`

**Review by JohnSchmeichel — CHANGES_REQUESTED** [review]

File: helpers/installcredprovider.sh [hunk 1]
```diff
@@ -12,11 +12,11 @@ REPO="Microsoft/artifacts-credprovider"
 NUGET_PLUGIN_DIR="$HOME/.nuget/plugins"
 
 # determine whether we install default or Net6
-if [[ -z ${USE_NET6_ARTIFACTS_CREDENTIAL_PROVIDER} ]] || [[ ${USE_NET6_ARTIFACTS_CREDENTIAL_PROVIDER} != "false" ]]; then
+if [ -z ${USE_NET6_ARTIFACTS_CREDENTIAL_PROVIDER} ] || [ ${USE_NET6_ARTIFACTS_CREDENTIAL_PROVIDER} != "false" ]; then
   FILE="Microsoft.Net6.NuGet.CredentialProvider.tar.gz"
 
   # throw if version starts with 0. (net6 not supported)
-  if [[ ! -z ${AZURE_ARTIFACTS_CREDENTIAL_PROVIDER_VERSION} ]] && [[ ${AZURE_ARTIFACTS_CREDENTIAL_PROVIDER_VERSION} == 0.* ]] || [[ ${AZURE_ARTIFACTS_CREDENTIAL_PROVIDER_VERSION} == v0.* ]]; then
+  if [ ! -z ${AZURE_ARTIFACTS_CREDENTIAL_PROVIDER_VERSION} ] && [ ${AZURE_ARTIFACTS_CREDENTIAL_PROVIDER_VERSION} == 0.* ] || [ "$AZURE_ARTIFACTS_CREDENTIAL_PROVIDER_VERSION" = "v0.*" ]; then
```
[the remark this answers is not in the document]
Comment (JohnSchmeichel) [reply helpers/installcredprovider.sh:19]: Think we can convert this to a case statement instead, which should work for both and simplify things:

```sh
  case ${AZURE_ARTIFACTS_CREDENTIAL_PROVIDER_VERSION} in
    0.*|v0.*)
        echo "ERROR: To install NET6 cred provider using the USE_NET6_ARTIFACTS_CREDENTIAL_PROVIDER variable, version to be installed must be 1.0.0. or greater. Check your AZURE_ARTIFACTS_CREDENTIAL_PROVIDER_VERSION variable."
        exit 1
        ;;
  esac
```

**Comment by phil-hodgson** [comment]
If we're making this work for environments without bash, should we also update the file's [shebang](https://en.wikipedia.org/wiki/Shebang_(Unix))?

**Comment by phil-hodgson** [comment]
The issue customers faced is that on some environments, the customer had to first install bash before they could get the cred provider installed. The installation script should work whether bash is available or not.

File: helpers/installcredprovider.sh (hunk 1 above)
[the remark this answers is not in the document]
Comment (amp-powell) [reply helpers/installcredprovider.sh]: Made these changes!

File: helpers/installcredprovider.sh [hunk 2]
```diff
@@ -16,9 +16,13 @@ if [ -z ${USE_NET6_ARTIFACTS_CREDENTIAL_PROVIDER} ] || [ ${USE_NET6_ARTIFACTS_CR
   FILE="Microsoft.Net6.NuGet.CredentialProvider.tar.gz"
 
   # throw if version starts with 0. (net6 not supported)
-  if [ ! -z ${AZURE_ARTIFACTS_CREDENTIAL_PROVIDER_VERSION} ] && [ ${AZURE_ARTIFACTS_CREDENTIAL_PROVIDER_VERSION} == 0.* ] || [ "$AZURE_ARTIFACTS_CREDENTIAL_PROVIDER_VERSION" = "v0.*" ]; then
-    echo "ERROR: To install NET6 cred provider using the USE_NET6_ARTIFACTS_CREDENTIAL_PROVIDER variable, version to be installed must be 1.0.0. or greater. Check your AZURE_ARTIFACTS_CREDENTIAL_PROVIDER_VERSION variable."
-    exit 1
+  if [ ! -z ${AZURE_ARTIFACTS_CREDENTIAL_PROVIDER_VERSION} ]; then
```
Comment (JohnSchmeichel) [inline helpers/installcredprovider.sh:19]: you should be able to remove this entirely, as if it's null it will just drop through the case

**Review by JohnSchmeichel — APPROVED** [review]

**Pull request merged**